### PR TITLE
FRDQ-114 - Testing useOutsideClick and useEventListener hooks.

### PR DESCRIPTION
**Pull request by Krysik**

### CODE REVIEW

**Comment by hayuna** [comment]
To avoid duplicates in import, can we move hook file into index?

Then import will looks like this `hooks/useHook`
instead of `hooks/useHook/useHook`

**Comment by hayuna** [comment]
useEventListener is not fully covered. Could you take a look into this?

File: src/hooks/useOutsideClick/useOutsideClick.test.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React, { useRef } from 'react';
+import { fireEvent, render } from '@testing-library/react';
+import useOutsideClick from './useOutsideClick';
+
+const handler = jest.fn();
+function ClickableComponent(): JSX.Element {
```
Comment (hayuna) [inline src/hooks/useOutsideClick/useOutsideClick.test.tsx:6]: We are working in ES6+, Could we change it into functional component using ES6 syntax?

File: src/hooks/useOutsideClick/useOutsideClick.test.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  const ref = useRef<HTMLDivElement>(null);
+  useOutsideClick(ref, handler);
+  return <div id="clickable" ref={ref} data-testid="clickable" />;
```
Comment (hayuna) [inline src/hooks/useOutsideClick/useOutsideClick.test.tsx:9]: I'm not sure we are using id and data-testid in this place. Can we remove it?

File: src/hooks/useOutsideClick/useOutsideClick.test.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+function Container() {
+  return (
+    <div data-testid="dummy">
+      <ClickableComponent />
+    </div>
+  );
+}
+
+describe('useOutsideClick hook', () => {
+  test('should call event handler function', async () => {
```
Comment (hayuna) [inline src/hooks/useOutsideClick/useOutsideClick.test.tsx:21]: Which method is asynchronyous here?

**Review by hayuna — CHANGES_REQUESTED** [review]
Comments above

File: src/hooks/useOutsideClick/useOutsideClick.test.tsx (hunk 3 above)
Comment (Krysik) [reply src/hooks/useOutsideClick/useOutsideClick.test.tsx:21]: I just forgot to remove it

**Pull request merged**